### DISCUSSION TITLE
Support for Docker Secrets

### DISCUSSION THREAD

**Discussion by Radiergummi** [lowdefy/lowdefy#1155] [Ideas and feature requests]
Hey there! I'm just migrating some of our applications into a [Docker Swarm mode](https://docs.docker.com/engine/swarm/) cluster, including Lowdefy. Due to that change, I've switched from environment configuration to [Docker Secrets](https://docs.docker.com/engine/swarm/secrets/). So what we're doing is set all configuration as repository variables on GitHub, and set them as secrets during the CI deployment action:
```bash
#!/bin/sh
set -e

vars=$(printenv | sed -n 's/^APP_\(.[^=]*\)=.*/\1/p')
for v in $vars; do
    n="APP_${v}"
    echo "${!n}" > "$(echo ${v} | tr '[:upper:]' '[:lower:]').txt"
done
```
So that takes `APP_FOO` from the environment and creates `./foo.txt` in the working directory. The docker-compose file picks that up and sets a secret:
```yaml
secrets:
    foo:
        file: foo.txt
```

As Lowdefy currently does not support secrets directly, I created a custom entrypoint script to set all secrets as environment variables:
```bash
#!/bin/sh
set -e

if [ -d "/run/secrets" ]; then
    for secret in /run/secrets/*; do
        secret_name=$(basename "${secret}" | tr "[:lower:]" "[:upper:]")
        variable_name="LOWDEFY_SECRET_${secret_name}"
        value=$(cat "${secret}")

        export "${variable_name}"="${value}"
    done
fi

exec /usr/local/bin/docker-entrypoint.sh "$@"
```
This takes `/run/secret/foo` and sets an environment variable `LODEFY_SECRET_FOO` with the file content. It works, but I'd obviously prefer native support for secrets.  
You're probably knee-deep in releasing v4 right now, but I was wondering if there are any plans for supporting secrets yet? Otherwise, I might take a stab at it once v4 is out 🙂

**Comment by Gervwyk** [comment]
Thanks for sharing @Radiergummi. It would be great to add official support for this!
For now, our focus is on getting everything v4 stable. Which has been delayed a little bit by customer implementation work. But we should be free to focus on the last legs of v4 in the next sprint.
